python/mmSolver/tools/averagemarker/tool.py

```diff
@@ -0,0 +1,97 @@
+"""
+This tool averages markers position from selected markers.
+"""
+
+import maya.cmds
+import mmSolver.tools.selection.filternodes as filternodes
+import mmSolver.logger
+import mmSolver.tools.createmarker.tool
+import maya.mel
+import mmSolver.api as mmapi
+
+LOG = mmSolver.logger.get_logger()
+
+
+def average_marker():
+    """
+    Averages marker position from selected markers.
+    :return: None
+    """
+
+    selection = maya.cmds.ls(sl=True)
+
+    selected_markers = filternodes.get_marker_nodes(selection)
+
+    if not selected_markers:
+        LOG.warning('Please select more than 2 markers ')
+        return
+
+    if len(selected_markers) < 2:
+        LOG.warning('Please select more than 2 markers')
+        return
+
+    mkr_selection = selected_markers[0]
+    mkr = mmapi.Marker(name=mkr_selection)
+    # getting camera from the selected marker
+    cam_from_mkr = mkr.get_camera()
+    mkr_name = mmapi.get_marker_name('marker1')
+    new_mkr = mmapi.Marker().create_node(cam=cam_from_mkr,
+                                         name=mkr_name)
+
+    new_mkr_node = new_mkr.get_node()
+    bnd_name = mmapi.get_bundle_name('bundle1')
+    new_bnd = mmapi.Bundle().create_node(name=bnd_name)
+    # connecting bundle to the marker
+    new_mkr.set_bundle(new_bnd)
+    # getting first frame and last frame from the selected markers
+    first_frame = []
+    last_frame = []
+    for marker in selected_markers:
+        plugs = [
+            '%s.translateX' % marker,
+            '%s.translateY' % marker,
+        ]
+        plug_lock_state = {}
+        for plug_name in plugs:
+            value = maya.cmds.getAttr(plug_name, lock=True)
+            plug_lock_state[plug_name] = value
+            maya.cmds.setAttr(plug_name, lock=False)
```

Comment:
I don't think we need to unlock and lock an attribute to query it.
Are you sure we need to unlock and relock?

If not, please remove this section.

Reply:
As discussed, if Maya needs to have the attributes unlocked to query the keyframes, that is unfortunate, however I expect we can work around it. I really don't want to unlock/relock attributes if we don't need to.

I'd expect we won't need to use the attribute, but rather get the animCurve that's connected to it.
We can get the animCurve with `maya.cmds.listConnections`, like this:
```
plug = 'node.translateX'
animCurves = maya.cmds.listConnections(plug, type='animCurve') or []
if len(animCurves) > 0:
    frame_num = maya.cmds.findKeyframe(animCurves[0], which='first')
```

Alternatively, rather than ``maya.cmds.findKeyframe(animCurve, which='first')``, we think we can use ``times = maya.cmds.keyframe(node, query=True, timeChange=True)``, and then we can get the first and last numbers in ``times``.